pymc3/step_methods/hmc.py

```diff
@@ -92,7 +93,12 @@ def astep(self, q0, logp, dlogp):
         return metrop_select(mr, q, q0)
 
 
-    
+    @staticmethod
+    def competence(var):
+        if var in discrete_types:
+            return 0
```

Comment:
not necessarily, can just be a few globals:
`DISCRETE = 0` to enhance readability.